lib/array_intersection.rb

```diff
@@ -1,6 +1,31 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), because the lookup for a hash is constant, and the times we do a lookup depends on the length of the longer array
```

Comment:
I believe this is an implementation of insertion sort, which is quadratic in time complexity.
Source 1: [Comparison Sort](https://en.wikipedia.org/wiki/Sorting_algorithm#Comparison_sorts)
Source 2: [Time Complexity](https://en.wikipedia.org/wiki/Time_complexity#Sub-quadratic_time)